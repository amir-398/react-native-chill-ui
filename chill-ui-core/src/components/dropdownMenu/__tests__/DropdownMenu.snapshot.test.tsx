import React from 'react';
import { render } from '@testing-library/react-native';

// Mock all the components we need
function DropdownMenu({ children, ...props }: any) {
  return (
    <div data-testid="dropdown-menu" {...props}>
      {children}
    </div>
  );
}

function DropdownMenuTrigger({ children, ...props }: any) {
  return (
    <div data-testid="dropdown-menu-trigger" {...props}>
      {children}
    </div>
  );
}

function DropdownMenuContent({ children, ...props }: any) {
  return (
    <div data-testid="dropdown-menu-content" {...props}>
      {children}
    </div>
  );
}

function DropdownMenuItem({ children, ...props }: any) {
  return (
    <div data-testid="dropdown-menu-item" {...props}>
      {children}
    </div>
  );
}

function DropdownMenuLabel({ children, ...props }: any) {
  return (
    <div data-testid="dropdown-menu-label" {...props}>
      {children}
    </div>
  );
}

function DropdownMenuSeparator({ ...props }: any) {
  return <div data-testid="dropdown-menu-separator" {...props} />;
}

// Mocks
jest.mock('react-native-reanimated', () => ({
  default: {
    View: ({ children, ...props }: any) => <div {...props}>{children}</div>,
  },
  runOnJS: jest.fn(),
  useAnimatedStyle: jest.fn(() => ({})),
  useSharedValue: jest.fn(() => ({ value: 0 })),
  withTiming: jest.fn(),
}));

jest.mock('../../../utils', () => ({
  classNameHandler: jest.fn(() => ({})),
  classNamePropsHandler: jest.fn(),
  cn: jest.fn((...args) => args.filter(Boolean).join(' ')),
  isNativeWindInstalled: jest.fn(() => false),
  isString: jest.fn(value => typeof value === 'string'),
  SlotSs: ({ children, ...props }: any) => <div {...props}>{children}</div>,
  SlotTw: ({ children, ...props }: any) => <div {...props}>{children}</div>,
  styleHandler: jest.fn(() => ({})),
}));

jest.mock('../../../components/box', () => ({
  Box: ({ children, ...props }: any) => <div {...props}>{children}</div>,
  BoxSs: ({ children, ...props }: any) => <div {...props}>{children}</div>,
  BoxTw: ({ children, ...props }: any) => <div {...props}>{children}</div>,
}));

jest.mock('../../../components/string', () => ({
  String: ({ children, ...props }: any) => <div {...props}>{children}</div>,
  StringSs: ({ children, ...props }: any) => <div {...props}>{children}</div>,
  StringTw: ({ children, ...props }: any) => <div {...props}>{children}</div>,
}));

jest.mock('../../../components/inputDropdown', () => ({
  InputDropdownModal: ({ children, ...props }: any) => <div {...props}>{children}</div>,
  InputDropdownModalSs: ({ children, ...props }: any) => <div {...props}>{children}</div>,
  InputDropdownModalTw: ({ children, ...props }: any) => <div {...props}>{children}</div>,
}));

jest.mock('../components/DropdownMenuContext', () => ({
  DropdownMenuProvider: ({ children }: any) => children,
  useDropdownMenuContext: () => ({
    dropdownRef: { current: null },
    dropdownStyles: {},
    eventClose: jest.fn(),
    eventOpen: jest.fn(),
    handleSelectItem: jest.fn(),
    inputRef: { current: null },
    setDropdownPosition: jest.fn(),
    setDropdownWidth: jest.fn(),
    setOffsets: jest.fn(),
    toggleDropdown: jest.fn(),
    visible: false,
    wrapperRef: { current: null },
  }),
}));

jest.mock('../styles/DropdownMenu.ss.styles', () => ({
  dropdownMenuItemSv: jest.fn(() => ({})),
  styles: {
    dropdown: {},
    item: {},
    label: {},
    separator: {},
    trigger: {},
  },
}));

jest.mock('../styles/DropdownMenu.tw.styles', () => ({
  dropdownMenuItemTv: jest.fn(() => ''),
  twStyles: {
    dropdown: '',
    item: '',
    label: '',
    separator: '',
    trigger: '',
  },
}));

jest.mock('../utils/defaultProps', () => ({
  dropdownMenuContentDefaultProps: {
    align: 'start',
    closeWhenInteractedOutside: true,
    offsetX: 0,
    offsetY: 5,
    side: 'bottom',
  },
  dropdownMenuDefaultProps: {
    defaultOpen: false,
  },
  dropdownMenuItemDefaultProps: {
    as: 'touchable-highlight',
    closeOnSelect: true,
    disabled: false,
    underlayColor: '#f0f0f0',
  },
  dropdownMenuLabelDefaultProps: {},
  dropdownMenuSeparatorDefaultProps: {},
  dropdownMenuTriggerDefaultProps: {
    as: 'pressable',
  },
}));

describe('DropdownMenu Snapshot Tests', () => {
  it('should match snapshot for basic dropdown', () => {
    const tree = render(
      <DropdownMenu>
        <DropdownMenuTrigger>
          <button>Open Menu</button>
        </DropdownMenuTrigger>
        <DropdownMenuContent>
          <DropdownMenuItem>Item 1</DropdownMenuItem>
          <DropdownMenuItem>Item 2</DropdownMenuItem>
        </DropdownMenuContent>
      </DropdownMenu>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot for controlled dropdown', () => {
    const tree = render(
      <DropdownMenu open onOpenChange={jest.fn()}>
        <DropdownMenuTrigger>
          <button>Open Menu</button>
        </DropdownMenuTrigger>
        <DropdownMenuContent>
          <DropdownMenuItem>Item 1</DropdownMenuItem>
        </DropdownMenuContent>
      </DropdownMenu>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot for dropdown with all components', () => {
    const tree = render(
      <DropdownMenu>
        <DropdownMenuTrigger as="touchable-opacity">
          <button>Open Menu</button>
        </DropdownMenuTrigger>
        <DropdownMenuContent width={300} offsetX={10} offsetY={5} verticalPosition="top" horizontalPosition="left">
          <DropdownMenuLabel>My Account</DropdownMenuLabel>
          <DropdownMenuItem as="pressable" onSelect={jest.fn()}>
            Profile
          </DropdownMenuItem>
          <DropdownMenuItem as="touchable-highlight" underlayColor="#e0e0e0">
            Settings
          </DropdownMenuItem>
          <DropdownMenuSeparator />
          <DropdownMenuItem disabled>Disabled Item</DropdownMenuItem>
        </DropdownMenuContent>
      </DropdownMenu>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot for dropdown with asChild components', () => {
    const tree = render(
      <DropdownMenu>
        <DropdownMenuTrigger asChild>
          <button>Custom Trigger</button>
        </DropdownMenuTrigger>
        <DropdownMenuContent>
          <DropdownMenuLabel asChild>
            <div>Custom Label</div>
          </DropdownMenuLabel>
          <DropdownMenuItem asChild closeOnSelect={false}>
            <button>Custom Item</button>
          </DropdownMenuItem>
        </DropdownMenuContent>
      </DropdownMenu>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot for dropdown with custom styling', () => {
    const tree = render(
      <DropdownMenu>
        <DropdownMenuTrigger className="custom-trigger">
          <button>Styled Trigger</button>
        </DropdownMenuTrigger>
        <DropdownMenuContent
          className="w-64 rounded-lg bg-white shadow-lg"
          style={{ backgroundColor: 'white', borderRadius: 8 }}
        >
          <DropdownMenuLabel className="font-semibold text-gray-500">Styled Label</DropdownMenuLabel>
          <DropdownMenuItem className="hover:bg-gray-100" style={{ padding: 12 }}>
            Styled Item
          </DropdownMenuItem>
        </DropdownMenuContent>
      </DropdownMenu>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot for dropdown with all positioning options', () => {
    const tree = render(
      <DropdownMenuContent
        width={400}
        offsetX={20}
        offsetY={10}
        maxHeight={300}
        minHeight={100}
        hasScroll
        hasAnimation
        closeDropdownWhenSelectedItem={false}
        closeWhenInteractedOutside={false}
        verticalPosition="bottom"
        horizontalPosition="right"
      >
        <DropdownMenuItem>Positioned Item</DropdownMenuItem>
      </DropdownMenuContent>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot for dropdown with all item variants', () => {
    const tree = render(
      <DropdownMenuContent>
        <DropdownMenuItem as="pressable" closeOnSelect>
          Pressable Item
        </DropdownMenuItem>
        <DropdownMenuItem as="touchable-opacity" closeOnSelect={false}>
          TouchableOpacity Item
        </DropdownMenuItem>
        <DropdownMenuItem as="touchable-highlight" underlayColor="#f0f0f0">
          TouchableHighlight Item
        </DropdownMenuItem>
        <DropdownMenuItem disabled>Disabled Item</DropdownMenuItem>
      </DropdownMenuContent>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot for empty dropdown', () => {
    const tree = render(
      <DropdownMenu>
        <DropdownMenuTrigger>
          <button>Empty Menu</button>
        </DropdownMenuTrigger>
        <DropdownMenuContent>{/* Empty content */}</DropdownMenuContent>
      </DropdownMenu>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot for dropdown with string children', () => {
    const tree = render(
      <DropdownMenuContent>
        <DropdownMenuLabel>String Label</DropdownMenuLabel>
        <DropdownMenuItem>String Item</DropdownMenuItem>
        <DropdownMenuSeparator />
      </DropdownMenuContent>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot for complex nested dropdown', () => {
    const tree = render(
      <DropdownMenu defaultOpen>
        <DropdownMenuTrigger as="touchable-highlight">
          <button>Complex Menu</button>
        </DropdownMenuTrigger>
        <DropdownMenuContent
          width={500}
          offsetX={15}
          offsetY={8}
          maxHeight={400}
          hasScroll
          closeWhenInteractedOutside
          verticalPosition="auto"
          horizontalPosition="auto"
        >
          <DropdownMenuLabel>Account Management</DropdownMenuLabel>
          <DropdownMenuItem as="pressable" onSelect={jest.fn()}>
            View Profile
          </DropdownMenuItem>
          <DropdownMenuItem as="touchable-opacity" onSelect={jest.fn()}>
            Edit Profile
          </DropdownMenuItem>
          <DropdownMenuSeparator />
          <DropdownMenuLabel>Settings</DropdownMenuLabel>
          <DropdownMenuItem as="touchable-highlight" underlayColor="#e0e0e0" onSelect={jest.fn()}>
            Preferences
          </DropdownMenuItem>
          <DropdownMenuItem asChild closeOnSelect={false}>
            <button>Custom Action</button>
          </DropdownMenuItem>
          <DropdownMenuSeparator />
          <DropdownMenuItem disabled>Premium Features (Coming Soon)</DropdownMenuItem>
          <DropdownMenuItem onSelect={jest.fn()}>Sign Out</DropdownMenuItem>
        </DropdownMenuContent>
      </DropdownMenu>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
