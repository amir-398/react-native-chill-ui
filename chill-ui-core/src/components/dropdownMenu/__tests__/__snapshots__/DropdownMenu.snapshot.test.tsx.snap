// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu Snapshot Tests should match snapshot for basic dropdown 1`] = `
<div
  data-testid="dropdown-menu"
>
  <div
    data-testid="dropdown-menu-trigger"
  >
    <button>
      Open Menu
    </button>
  </div>
  <div
    data-testid="dropdown-menu-content"
  >
    <div
      data-testid="dropdown-menu-item"
    >
      Item 1
    </div>
    <div
      data-testid="dropdown-menu-item"
    >
      Item 2
    </div>
  </div>
</div>
`;

exports[`DropdownMenu Snapshot Tests should match snapshot for complex nested dropdown 1`] = `
<div
  data-testid="dropdown-menu"
  defaultOpen={true}
>
  <div
    as="touchable-highlight"
    data-testid="dropdown-menu-trigger"
  >
    <button>
      Complex Menu
    </button>
  </div>
  <div
    closeWhenInteractedOutside={true}
    data-testid="dropdown-menu-content"
    hasScroll={true}
    horizontalPosition="auto"
    maxHeight={400}
    offsetX={15}
    offsetY={8}
    verticalPosition="auto"
    width={500}
  >
    <div
      data-testid="dropdown-menu-label"
    >
      Account Management
    </div>
    <div
      as="pressable"
      data-testid="dropdown-menu-item"
      onSelect={[MockFunction]}
    >
      View Profile
    </div>
    <div
      as="touchable-opacity"
      data-testid="dropdown-menu-item"
      onSelect={[MockFunction]}
    >
      Edit Profile
    </div>
    <div
      data-testid="dropdown-menu-separator"
    />
    <div
      data-testid="dropdown-menu-label"
    >
      Settings
    </div>
    <div
      as="touchable-highlight"
      data-testid="dropdown-menu-item"
      onSelect={[MockFunction]}
      underlayColor="#e0e0e0"
    >
      Preferences
    </div>
    <div
      asChild={true}
      closeOnSelect={false}
      data-testid="dropdown-menu-item"
    >
      <button>
        Custom Action
      </button>
    </div>
    <div
      data-testid="dropdown-menu-separator"
    />
    <div
      data-testid="dropdown-menu-item"
      disabled={true}
    >
      Premium Features (Coming Soon)
    </div>
    <div
      data-testid="dropdown-menu-item"
      onSelect={[MockFunction]}
    >
      Sign Out
    </div>
  </div>
</div>
`;

exports[`DropdownMenu Snapshot Tests should match snapshot for controlled dropdown 1`] = `
<div
  data-testid="dropdown-menu"
  onOpenChange={[MockFunction]}
  open={true}
>
  <div
    data-testid="dropdown-menu-trigger"
  >
    <button>
      Open Menu
    </button>
  </div>
  <div
    data-testid="dropdown-menu-content"
  >
    <div
      data-testid="dropdown-menu-item"
    >
      Item 1
    </div>
  </div>
</div>
`;

exports[`DropdownMenu Snapshot Tests should match snapshot for dropdown with all components 1`] = `
<div
  data-testid="dropdown-menu"
>
  <div
    as="touchable-opacity"
    data-testid="dropdown-menu-trigger"
  >
    <button>
      Open Menu
    </button>
  </div>
  <div
    data-testid="dropdown-menu-content"
    horizontalPosition="left"
    offsetX={10}
    offsetY={5}
    verticalPosition="top"
    width={300}
  >
    <div
      data-testid="dropdown-menu-label"
    >
      My Account
    </div>
    <div
      as="pressable"
      data-testid="dropdown-menu-item"
      onSelect={[MockFunction]}
    >
      Profile
    </div>
    <div
      as="touchable-highlight"
      data-testid="dropdown-menu-item"
      underlayColor="#e0e0e0"
    >
      Settings
    </div>
    <div
      data-testid="dropdown-menu-separator"
    />
    <div
      data-testid="dropdown-menu-item"
      disabled={true}
    >
      Disabled Item
    </div>
  </div>
</div>
`;

exports[`DropdownMenu Snapshot Tests should match snapshot for dropdown with all item variants 1`] = `
<div
  data-testid="dropdown-menu-content"
>
  <div
    as="pressable"
    closeOnSelect={true}
    data-testid="dropdown-menu-item"
  >
    Pressable Item
  </div>
  <div
    as="touchable-opacity"
    closeOnSelect={false}
    data-testid="dropdown-menu-item"
  >
    TouchableOpacity Item
  </div>
  <div
    as="touchable-highlight"
    data-testid="dropdown-menu-item"
    underlayColor="#f0f0f0"
  >
    TouchableHighlight Item
  </div>
  <div
    data-testid="dropdown-menu-item"
    disabled={true}
  >
    Disabled Item
  </div>
</div>
`;

exports[`DropdownMenu Snapshot Tests should match snapshot for dropdown with all positioning options 1`] = `
<div
  closeDropdownWhenSelectedItem={false}
  closeWhenInteractedOutside={false}
  data-testid="dropdown-menu-content"
  hasAnimation={true}
  hasScroll={true}
  horizontalPosition="right"
  maxHeight={300}
  minHeight={100}
  offsetX={20}
  offsetY={10}
  verticalPosition="bottom"
  width={400}
>
  <div
    data-testid="dropdown-menu-item"
  >
    Positioned Item
  </div>
</div>
`;

exports[`DropdownMenu Snapshot Tests should match snapshot for dropdown with asChild components 1`] = `
<div
  data-testid="dropdown-menu"
>
  <div
    asChild={true}
    data-testid="dropdown-menu-trigger"
  >
    <button>
      Custom Trigger
    </button>
  </div>
  <div
    data-testid="dropdown-menu-content"
  >
    <div
      asChild={true}
      data-testid="dropdown-menu-label"
    >
      <div>
        Custom Label
      </div>
    </div>
    <div
      asChild={true}
      closeOnSelect={false}
      data-testid="dropdown-menu-item"
    >
      <button>
        Custom Item
      </button>
    </div>
  </div>
</div>
`;

exports[`DropdownMenu Snapshot Tests should match snapshot for dropdown with custom styling 1`] = `
<div
  data-testid="dropdown-menu"
>
  <div
    className="custom-trigger"
    data-testid="dropdown-menu-trigger"
  >
    <button>
      Styled Trigger
    </button>
  </div>
  <div
    className="w-64 rounded-lg bg-white shadow-lg"
    data-testid="dropdown-menu-content"
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 8,
      }
    }
  >
    <div
      className="font-semibold text-gray-500"
      data-testid="dropdown-menu-label"
    >
      Styled Label
    </div>
    <div
      className="hover:bg-gray-100"
      data-testid="dropdown-menu-item"
      style={
        {
          "padding": 12,
        }
      }
    >
      Styled Item
    </div>
  </div>
</div>
`;

exports[`DropdownMenu Snapshot Tests should match snapshot for dropdown with string children 1`] = `
<div
  data-testid="dropdown-menu-content"
>
  <div
    data-testid="dropdown-menu-label"
  >
    String Label
  </div>
  <div
    data-testid="dropdown-menu-item"
  >
    String Item
  </div>
  <div
    data-testid="dropdown-menu-separator"
  />
</div>
`;

exports[`DropdownMenu Snapshot Tests should match snapshot for empty dropdown 1`] = `
<div
  data-testid="dropdown-menu"
>
  <div
    data-testid="dropdown-menu-trigger"
  >
    <button>
      Empty Menu
    </button>
  </div>
  <div
    data-testid="dropdown-menu-content"
  />
</div>
`;
